### PR TITLE
BAVL-138 intial commit containing groundwork for pulling back locations via the locations API. Missing part is actual call to locations API.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
As part of this change it includes the removal of the prison description field as this is deemed redundant.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt [hunk 1]
```diff
@@ -22,6 +22,9 @@ class LocationsInsidePrisonClient(private val locationsInsidePrisonApiWebClient:
     .bodyToMono(typeReference<List<Location>>())
     .onErrorResume(WebClientResponseException.NotFound::class.java) { Mono.empty() }
     .block() ?: emptyList()
+
+  // TODO to be implemented
+  fun getLocationsAtPrison(prisonCode: String): List<Location> = emptyList()
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt:27]: Need to wire in actual call to the locations API here.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.Location
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.LocationsService
+
+@Tag(name = "Locations Controller")
+@RestController
+@RequestMapping(value = ["locations"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class LocationsController(private val locationsService: LocationsService) {
+
+  @Operation(summary = "Endpoint to return a list of locations for a prison")
+  @ApiResponses(
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt:27]: List of appointment type / video type locations in a prison?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  @PreAuthorize("hasAnyRole('BOOK_A_VIDEO_LINK_ADMIN')")
+  fun prisonLocations(
+    @Parameter(description = "The prison code for which locations will be retrieved.")
+    @RequestParam(name = "prisonCode", required = true)
+    prisonCode: String,
+    @Parameter(description = "Enabled only, true or false. Defaults to false if not supplied.")
+    @RequestParam(name = "enabledOnly", required = false)
+    enabledOnly: Boolean = false,
+  ): List<Location> = locationsService.getLocationsAtPrison(prisonCode, enabledOnly)
+}
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt:71]: getVideo/AppointmentLocationsAtPrison?

**Review by timharrison-moj — APPROVED** [review]
LGTM. Minor comments only.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/helper/Prisons.kt [hunk 4]
```diff
@@ -4,3 +4,10 @@ const val BIRMINGHAM = "BMI"
 const val MOORLAND = "MDI"
 const val RISLEY = "RSI"
 const val WERRINGTON = "WNI"
+
+val prisonNames = mapOf(
+  BIRMINGHAM to "Birmingham",
+  MOORLAND to "Moorland",
+  RISLEY to "Risley",
+  WERRINGTON to "Werrington",
```
[the remark this answers is not in the document]
Comment (matt-crowson-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/helper/Prisons.kt:12]: So this helps with unit tests (not for integration tests).  Not strictly necessary but helps a little to differentiate in unit tests.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt (hunk 3 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt:71]: Will do in next PR.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt (hunk 2 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/LocationsController.kt:27]: Already done this locally so will be in next PR.

**Pull request merged**